machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +54,22 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
+	// GuestAccelerators is a list of GPUs to be attached to the VM.
+	// +optional
+	GuestAccelerators []*GCPAcceleratorConfig `json:"guestAccelerators,omitempty"`
 	// Preemptible indicates if created instance is preemptible
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// [default] MIGRATE - causes Compute Engine to live migrate an instance when there is a maintenance event.
+	// TERMINATE - stops an instance instead of migrating it.
```

Comment:
Enumerated values in kube APIs are "Terminate".  Also, you're missing the tags for CRD validation on this field.

Reply:
Should we change this to be a typed string and then add constants that match that type with the correct values? That way we don't have to spell them out in our controllers and tests multiple times.

@deads2k is making it a const like this idiomatic for enums?

```
type GCPHostMaintenanceType string

const (
  MigrateHostMaintenanceType GCPHostMaintenanceType = "Migrate"
  TerminateHostMaintenanceType GCPHostMaintenanceType = "Terminate"
)
```

We don't have CRD validation for the GCP provider spec (or any provider spec) due to it being passed in as a raw extension, so we could add the CRD validation tags but they unfortunately have no effect right now